src/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -297,26 +316,48 @@ class ThetaStarPathPlanner : public PathPlanner
     void resetAndInitializeMemberVariables(const Rectangle &navigable_area,
                                            const std::vector<ObstaclePtr> &obstacles);
 
-    // if close to destination then return no path
+    /**
+     * Returns a collision-free hash for the given Coordinate
+     *
+     * @param coord Coordinate to hash
+     *
+     * @return hash of the Coordinate
+     */
+    unsigned long hashCoordinate(const Coordinate &coord) const;
+
+    /**
+     * Returns a collision-free hash for the given a pair of Coordinate
+     *
+     * @param coord1 Coordinate 1 of pair to hash
+     * @param coord2 Coordinate 2 of pair to hash
+     *
+     * @return hash of the pair of Coordinates
+     */
+    unsigned long hashCoordinatePair(
+        const ThetaStarPathPlanner::Coordinate &coord1,
+        const ThetaStarPathPlanner::Coordinate &coord2) const;
+
+    // if close to end then return no path
     static constexpr double CLOSE_TO_DEST_THRESHOLD = 0.01;  // in metres
 
     // increase in threshold to reduce oscillation
-    static constexpr int BLOCKED_DESINATION_OSCILLATION_MITIGATION =
+    static constexpr unsigned int BLOCKED_DESINATION_OSCILLATION_MITIGATION =
         2;  // multiples of CLOSE_TO_DEST_THRESHOLD to ignore to control oscillation
 
-    // resolution for searching for unblocked point around a blocked destination
+    // resolution for searching for unblocked point around a blocked end
     static constexpr double BLOCKED_DESTINATION_SEARCH_RESOLUTION =
         50.0;  // number of fractions to divide 1m
 
     // only change this value
-    static constexpr int GRID_DIVISION_FACTOR = 1;  // the n in the O(n^2) algorithm :p
+    static constexpr unsigned int GRID_DIVISION_FACTOR =
+        1;  // the n in the O(n^2) algorithm :p
     // don't change this calculation
     const double SIZE_OF_GRID_CELL_IN_METERS =
```

Comment:
Why not define this value directly rather than by a proxy `division factor`?

Reply:
seems like a good idea to for cell divisions to divide the field evenly in fractions of a robot radius

Reply:
You're not guaranteed to get perfectly even division (ie. divison that fits perfectly in the field) though. Robot radius doesn't divide evenly into both field length and width.

Making stuff relative to a robot radius is intuitive to picture, but IMO is otherwise arbitrary so I don't see a huge benefit here. Directly setting grid size would also give us the ability to more finely tune performance vs path fidelity.

Reply:
ok changed it to set the size directly